core/src/main/java/com/fincity/saas/core/service/connection/email/EmailService.java

```diff
@@ -63,7 +64,7 @@ public Mono<Boolean> sendEmail(String appCode, String clientCode, List<String> a
 		return FlatMapUtil.flatMapMono(
 
 				() -> {
-					if (appCode != null && clientCode != null)
+					if (!StringUtil.isNullOrBlank(appCode) && !StringUtil.isNullOrBlank(clientCode))
 						return Mono.just(Tuples.of(appCode, clientCode));
 
 					return SecurityContextUtil.getUsersContextAuthentication()
```

Comment:
use the following code instead of this, so if appcode is available it will be considered. you don't have to give both appcode and client code.

> SecurityContextUtil.getUsersContextAuthentication()
						.map(e -> Tuples.of(
								CommonsUtil.nonNullValue(appCode.trim().length() == 0 ? null : appCode,
										e.getUrlAppCode()),
								CommonsUtil.nonNullValue(clientCode.trim().length() == 0 ? null : clientCode,
										e.getUrlClientCode())))